✨ feat [#1] - 주문 생성 internal API 및 주문 조회 API 구현

mummumm opened the pull request:
## ✨ 변경 타입
* [x] 신규 기능 추가/수정
* [ ] 버그 수정
* [x] 리팩토링
* [ ] 설정
* [ ] 비기능 (주석 등 기능에 영향을 주지 않음)

## ✅ 체크리스트

* [x] 코드 작성 가이드라인을 준수했는가?
* [x] 변경 사항에 대한 테스트를 완료했는가?
* [x] 문서 변경이 필요한 경우, 해당 내용을 반영했는가?

## 🚀 PR 내용
* **한 줄 요약**
 - 주문 생성(1차 - feignClient 없는 버전)
 - 주문 조회 

* **세부 내용**
 - 주문 생성 HTTP 테스트 확인
 - 주문 조회 HTTP 테스트 확인

## 💡 추가 설명

* 코드 리뷰 시 특별히 참고해야 할 부분이 있다면 언급해주세요.
* 궁금한 점이나 논의하고 싶은 내용이 있다면 작성해주세요.

## 📚 참고 자료 (선택 사항)

* 관련 자료 링크

mummumm committed:
✨ feat[#1] : Order API Entity 및 기본 설정파일 추가
- properties 파일 삭제 및 yml파일 생성
- Order 관련 Entity 파일 생성

mummumm committed:
♻️ refactor [#1] : 테스트를 위해 Entity, 기본 설정파일 수정
및 공통모듈 버전 수정

mummumm committed:
✨ feat [#1] : 주문 생성, 주문 조회 api 구현 및 httpTest 작성

mummumm committed:
♻️ refactor [#1] : 공통모듈 Auditing 설정 추가

chokyuseong00 commented on src/main/resources/properties/datasource.yml:
```diff
@@ -0,0 +1,6 @@
+  datasource:
+    url: jdbc:postgresql://localhost:5432/test
+    username: test
+    password: 1234
```
민감한 정보는 플레이스 홀더를 적용하여 환경변수 처리하시는 방향도 좋을 것 같아요!

mummumm replied:
넵 !!

etuhcarap commented on src/main/java/com/sparta/limited/order_service/application/service/OrderServiceImpl.java:
```diff
@@ -0,0 +1,35 @@
+public class OrderServiceImpl implements OrderService {
+
+    private final OrderRepository orderRepository;
+    private final OrderMapper orderMapper;
```
OrderMapper가 빈이 아니므로 OrderServiceImpl이 더이상 OrderMapper에 대한 의존성 주입을 받지 않아도 괜찮을 것 같아요!

etuhcarap commented on src/main/java/com/sparta/limited/order_service/domain/model/Order.java:
```diff
@@ -0,0 +1,63 @@
+    @Column(nullable = false)
+    private Long userId;
+
+    @Enumerated(EnumType.STRING)
```
👍

etuhcarap commented on src/main/java/com/sparta/limited/order_service/domain/model/Order.java:
```diff
@@ -0,0 +1,63 @@
+        this.price = price;
+    }
+
+    public static Order of(Long userId, OrderType orderType, UUID productId,
```
정적 팩터리 메서드를 이용한 객체 생성 유도 👍

etuhcarap commented on src/main/java/com/sparta/limited/order_service/domain/model/OrderType.java:
```diff
@@ -0,0 +1,6 @@
+public enum OrderType {
+    AUCTION,
+    LIMITED
+}
```
주문의 각 상태값을 필드로 상세하게 표현해서 가독성과 상태 코드 파악에 용이하도록 구성하면 어떨까요? 

```java
public enum OrderType {
    AUCTION("경매?"),
    LIMITED("품절 ? 경매 완료?"),
    ;
    
    private final description;

    // 생성자..
}
```

etuhcarap commented on src/main/java/com/sparta/limited/order_service/infrastructure/persistence/JpaOrderRepository.java:
```diff
@@ -0,0 +1,11 @@
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+@Repository
```
interface는 인스턴스가 생성되지 않기 때문에 빈 스캔을 위한 @Repository와 같은 어노테이션을 명시하지 않아도 괜찮아요!

etuhcarap commented on src/main/resources/application.yml:
```diff
@@ -0,0 +1,8 @@
+  config:
+    import:
+      - classpath:properties/datasource.yml
+      - classpath:properties/jpa.yml
```
Application이 운영되면서 기동에 필요한 다양한 설정들이 추가되어 application.yml이 복잡해질 수 있어요,
spring은 multi-profile을 지원하므로, test/local/dev/stg/sandbox/prd 와 같은 많은 실행 환경이 구성될 수록 application은 점점더 비대해지고 복잡해질 수 있어요!

그런 의미에서 분류별로 properties를 잘 분리하셨어요! 👍 👍

AnJungGeon replied:
참고하겠습니다!

etuhcarap commented on src/test/resources/httpTest/OrderReadTest.http:
```diff
@@ -0,0 +1,7 @@
+
+### 주문 조회 (실패)
+GET localhost:8080/api/v1/orders/e53a2969-a914-4e7b-96d4-371799cef98c1
+Content-Type: application/json
```
굿굿
보다 복잡한 MSA 환경에서 빠르게 실행하고 확인할 수 있는 환경을 구성하는 것은 매우 중요해요! 👍 👍

etuhcarap commented on src/main/java/com/sparta/limited/order_service/infrastructure/repository/OrderRepositoryImpl.java:
```diff
@@ -0,0 +1,26 @@
+    @Override
+    public Optional<Order> findById(UUID id) {
+        return jpaOrderRepository.findById(id);
+    }
```
DB 접근 계층인 Repository에 관심을 가지는 Service 혹은 메서드가 많아질 수록 호출하는 쪽에서 존재하지 않는 주문에 대한 예외처리를 매번 구성해야 할 수 있을 것 같아요!

코드 중복을 줄이고 일관된 형태의 동작을 제공 할 수 있도록 예외처리를 동반하면 어떨까요?

```java
@Override
public Order findById(UUID id) {
    return jpaOrderRepository.findById(id)
            .orElseThrow(() -> new SomeOrderCustomException(SOME_ORDER_NOT_EXIST_ERROR_CODE));
}
```

mummumm replied:
감사합니다 적용해보겠습니다 !

etuhcarap reviewed (commented):
안녕하세요 은진님!
참고하실만한 내용을 리뷰로 남겨보았어요!

주말 늦게까지 같이 모여서 학습하시는 모습 너무 보기 좋아요!
환절기엔 항상 감기 조심하시고, 다음 PR도 화이팅!

AnJungGeon reviewed (approved):
수고하셨습니다

mummumm committed:
♻️ refactor [#1] : JpaOrderRepository 추가, repository에 OrderServiceImpl에서 사용할 공통 에러 관련 코드 추가 및 Enum에 타입별 설명 필드 추가
- JpaOrderRepository 추가로 도메인과 영속성 계층 분리
- OrderRepository 인터페이스 및 구현체 추가
- OrderStatus, OrderType에 설명 필드 추가

sohyun119 reviewed (approved):
튜터님께서 이미 너무 댓글을 잘 달아주셔서... 따로 달지 않고 승인하고 갈게용!
은진님 10000점..👍
